### PR TITLE
refactor: apply updated WidgetFrame and widget hook to CostMapWidget

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

- Moved dashboard scope components to upper scope.
- Removed store usages from WidgetFrame.
- Updated use-widget-frame, use-cost-widget hooks to return widgetFrameEventHandlers.
- Moved DashboardWidgetEditModal to DashboardWIdgetContainer.

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/_components/WidgetFrame.vue [hunk 1]
```diff
@@ -79,10 +80,17 @@ const props = withDefaults(defineProps<WidgetFrameProps>(), {
     nonInheritOptionsTooltipText: undefined,
 });
 
-const emit = defineEmits(['refresh']);
+// TODO: remove 'refresh' event after the widgets are refactored
+const emit = defineEmits<{(event: 'refresh'): void;
+    (event: 'click-delete'): void;
+    (event: 'click-expand'): void;
+    (event: 'click-edit'): void;
+}>();
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_components/WidgetFrame.vue:88]: praise: 👍

File: apps/web/src/services/dashboards/widgets/_configs/config.ts [hunk 2]
```diff
@@ -247,8 +248,24 @@ export interface WidgetProps {
...
+    errorMode?: boolean;
+    disableViewMode?: boolean;
+    disableRefreshOnVariableChange?: boolean;
+    initiated?: boolean;
+    currencyRates?: CurrencyRates;
+    allReferenceTypeInfo: AllReferenceTypeInfo;
+    settings?: DashboardSettings;
+    variablesSchema?: DashboardVariablesSchema;
+    variables?: DashboardVariables;
+}
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_configs/config.ts:265]: praise: much better..

File: apps/web/src/services/dashboards/shared/dashboard-widget-container/DashboardWidgetContainer.vue [hunk 3]
```diff
@@ -148,6 +153,19 @@ defineExpose({
     refreshAllWidget,
 });
 
+/* widget edit modal */
+const widgetEditState = reactive({
+    visibleModal: false,
+    targetWidget: null as DashboardLayoutWidgetInfo|null,
```
Comment (yuda110) [inline apps/web/src/services/dashboards/shared/dashboard-widget-container/DashboardWidgetContainer.vue:180]: nitpick: How about `targetWidgetInfo`...?

File: apps/web/src/services/dashboards/widgets/_components/DashboardWidgetEditModal.vue [hunk 4]
```diff
@@ -18,7 +18,9 @@ interface Props {
     widgetConfigId: string;
 }
 interface EmitFn {
-    (e: string, value: string): void;
+    (e: 'update:visible', value: boolean): void;
+    (e: 'confirm'): void;
```
Comment (piggggggggy) [inline apps/web/src/services/dashboards/widgets/_components/DashboardWidgetEditModal.vue:22]: praise: 👍

**Review by piggggggggy — APPROVED** [review]
LGTM!
Share plz.. 
Anyway, I think I need more understanding of widgets.

**Review by yuda110 — APPROVED** [review]
LGTM!!! I checked carefully.

**Pull request merged**